util/misc.go

```diff
@@ -434,9 +439,22 @@ type SequenceTable interface {
 }
 
 // LoadTLSCertificates loads CA/KEY/CERT for special paths.
-func LoadTLSCertificates(ca, key, cert string) (tlsConfig *tls.Config, err error) {
+func LoadTLSCertificates(ca, key, cert string, autoTLS bool) (tlsConfig *tls.Config, autoReload bool, err error) {
+	autoReload = false
 	if len(cert) == 0 || len(key) == 0 {
-		return
+		if !autoTLS {
+			logutil.BgLogger().Warn("Automatic TLS Certificate creation is disabled", zap.Error(err))
+			return
+		}
+		autoReload = true
+		tempStoragePath := config.GetGlobalConfig().TempStoragePath
+		cert = filepath.Join(tempStoragePath, "/cert.pem")
+		key = filepath.Join(tempStoragePath, "/key.pem")
+		err = createTLSCertificates(cert, key)
+		if err != nil {
+			logutil.BgLogger().Warn("TLS Certificate creation failed", zap.Error(err))
```

Comment:
It looks like an error is being downgraded to a warning and then ignored. If someone set auto-tls, then that is what they want and this should be a loud error.

Reply:
This is in line with the soft-fail that is done when TLS certificates are configured. I opened https://github.com/pingcap/tidb/issues/26065 to change that some time ago. This could also be made into a failure instead of a warning as part of that, but for now I don't think we should change this behavior.

Also `auto-tls` is enabled by default, so someone would only need to set this if they for some reason don't want this automatic generation of TLS certificates.

Reply:
Any other objection?